clickhouse/types/type_parser.cpp

```diff
@@ -138,6 +210,8 @@ TypeParser::Token TypeParser::NextToken() {
                 return Token{Token::RPar, StringView(cur_++, 1)};
             case ',':
                 return Token{Token::Comma, StringView(cur_++, 1)};
+            case '\'':
+                return Token{Token::Quote, StringView(cur_++, 1)};
```

Comment:
I figure, it's better to do quote-to-quote consumption here and generating `Token::String` or some `Token::QuotedString` directly from here. The current approach feels like it somewhat leaks abstractions from lexer (this function) and parser `Parse()`.

Reply:
agree, updated implementation of `NextToken()`